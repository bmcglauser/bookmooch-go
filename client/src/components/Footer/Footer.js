import React from 'react';
import './Footer.scss';
import buttons from '../FooterButtons';

function whichButton (inputStr, props) {
  switch(inputStr) {
    case 'back':
      return <buttons.BackButton ctx={props} />;
    case 'circleAdd':
      return <buttons.CircleAddButton />;
    case 'search':
      return <buttons.SearchButton />;
    case 'userHome':
      return <buttons.UserHomeButton />;
    default:
      return <div />;
  }
}

export default function Footer (props) {
  const {leftBut, centerBut, rightBut, ctx} = props
  return (
    <div className="footer-grand-wrapper">
      <div className="small-button">
        {whichButton(leftBut, ctx)}
      </div>
      {whichButton(centerBut, ctx)}
      <div className="small-button">
        {whichButton(rightBut, ctx)}
      </div>
    </div>
  );
}