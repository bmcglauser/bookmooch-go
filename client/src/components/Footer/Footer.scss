@use '../../App' as *;

.footer-grand-wrapper {
  width: 100%;
  height: 100%;
  display: flex;
  align-items: flex-end;
  justify-content: space-between;
  color: $off-white;

  .small-button {
    width: 85px;
    display: flex;
    align-items: center;
    justify-content: center;
  }
}
