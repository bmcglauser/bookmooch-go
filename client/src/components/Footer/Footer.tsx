import React, { FunctionComponent } from 'react';
import './Footer.scss';
import buttons from '../FooterButtons';
import { RouteComponentProps } from 'react-router-dom';

function whichButton(
  inputStr?: string,
  props?: RouteComponentProps
): JSX.Element {
  if (inputStr === 'back' && props) {
    return <buttons.BackButton ctx={props} />;
  }
  switch (inputStr) {
    case 'circleAdd':
      return <buttons.CircleAddButton />;
    case 'search':
      return <buttons.SearchButton />;
    case 'userHome':
      return <buttons.UserHomeButton />;
    default:
      return <div />;
  }
}

type FooterProps = {
  leftBut?: string;
  centerBut?: string;
  rightBut?: string;
  ctx?: RouteComponentProps;
};

const Footer: FunctionComponent<FooterProps> = (props) => {
  return (
    <div className="footer-grand-wrapper">
      <div className="small-button">
        {whichButton(props.leftBut, props.ctx)}
      </div>
      {whichButton(props.centerBut, props.ctx)}
      <div className="small-button">
        {whichButton(props.rightBut, props.ctx)}
      </div>
    </div>
  );
};

export default Footer;
