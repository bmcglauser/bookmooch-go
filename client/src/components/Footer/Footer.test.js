import React from 'react';
import { render } from '@testing-library/react';

import Footer from './';
import { MemoryRouter } from 'react-router-dom';

describe('Welcome Text', () => {
  describe('should render based on props passed', () => {
    it('should render with two corner buttons', () => {
      const tree = render(
      <Footer 
        leftBut={'userHome'}
        centerBut={null}
        rightBut={'search'} />, {wrapper: MemoryRouter});
      expect(tree).toMatchSnapshot();
    });
    it('should render with only center button', () => {
      const tree = render(
      <Footer 
        leftBut={null}
        centerBut={'circleAdd'}
        rightBut={null} />, {wrapper: MemoryRouter});
      expect(tree).toMatchSnapshot();
    });
    it('should render with one left button', () => {
      const tree = render(
      <Footer 
        leftBut={'userHome'}
        centerBut={null}
        rightBut={null} />, {wrapper: MemoryRouter});
      expect(tree).toMatchSnapshot();
    });
    it('should render with one right button', () => {
      const tree = render(
      <Footer 
        leftBut={null}
        centerBut={null}
        rightBut={'search'} />, {wrapper: MemoryRouter});
      expect(tree).toMatchSnapshot();
    });
    it('should render with one left button and center button', () => {
      const tree = render(
      <Footer 
        leftBut={'userHome'}
        centerBut={'circleAdd'}
        rightBut={null} />, {wrapper: MemoryRouter});
      expect(tree).toMatchSnapshot();
    });
  });
});
