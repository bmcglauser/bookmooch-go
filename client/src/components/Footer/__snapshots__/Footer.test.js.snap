// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Text should render based on props passed should render with one left button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="footer-grand-wrapper"
      >
        <div
          class="small-button"
        >
          <a
            href="/profile/"
            style="text-decoration: none;"
          >
            <div
              class="user-button-wrapper"
            >
              <button
                class="user-button"
              />
              <p>
                User Home
              </p>
            </div>
          </a>
        </div>
        <div />
        <div
          class="small-button"
        >
          <div />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="footer-grand-wrapper"
    >
      <div
        class="small-button"
      >
        <a
          href="/profile/"
          style="text-decoration: none;"
        >
          <div
            class="user-button-wrapper"
          >
            <button
              class="user-button"
            />
            <p>
              User Home
            </p>
          </div>
        </a>
      </div>
      <div />
      <div
        class="small-button"
      >
        <div />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Welcome Text should render based on props passed should render with one left button and center button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="footer-grand-wrapper"
      >
        <div
          class="small-button"
        >
          <a
            href="/profile/"
            style="text-decoration: none;"
          >
            <div
              class="user-button-wrapper"
            >
              <button
                class="user-button"
              />
              <p>
                User Home
              </p>
            </div>
          </a>
        </div>
        <div
          class="circle-add-button-wrapper"
        >
          <a
            href="/findadd"
            style="text-decoration: none;"
          >
            <button
              class="the-button"
            >
              <div
                class="the-circle"
              />
              <div
                class="vertical-bar"
              />
              <div
                class="horizontal-bar"
              />
            </button>
          </a>
          <p
            class="label"
          >
            Add a Book
          </p>
        </div>
        <div
          class="small-button"
        >
          <div />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="footer-grand-wrapper"
    >
      <div
        class="small-button"
      >
        <a
          href="/profile/"
          style="text-decoration: none;"
        >
          <div
            class="user-button-wrapper"
          >
            <button
              class="user-button"
            />
            <p>
              User Home
            </p>
          </div>
        </a>
      </div>
      <div
        class="circle-add-button-wrapper"
      >
        <a
          href="/findadd"
          style="text-decoration: none;"
        >
          <button
            class="the-button"
          >
            <div
              class="the-circle"
            />
            <div
              class="vertical-bar"
            />
            <div
              class="horizontal-bar"
            />
          </button>
        </a>
        <p
          class="label"
        >
          Add a Book
        </p>
      </div>
      <div
        class="small-button"
      >
        <div />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Welcome Text should render based on props passed should render with one right button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="footer-grand-wrapper"
      >
        <div
          class="small-button"
        >
          <div />
        </div>
        <div />
        <div
          class="small-button"
        >
          <a
            href="/search"
            style="text-decoration: none;"
          >
            <div
              class="search-button-wrapper"
            >
              <button
                class="search-button"
              />
              <p>
                Find a Book
              </p>
            </div>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="footer-grand-wrapper"
    >
      <div
        class="small-button"
      >
        <div />
      </div>
      <div />
      <div
        class="small-button"
      >
        <a
          href="/search"
          style="text-decoration: none;"
        >
          <div
            class="search-button-wrapper"
          >
            <button
              class="search-button"
            />
            <p>
              Find a Book
            </p>
          </div>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Welcome Text should render based on props passed should render with only center button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="footer-grand-wrapper"
      >
        <div
          class="small-button"
        >
          <div />
        </div>
        <div
          class="circle-add-button-wrapper"
        >
          <a
            href="/findadd"
            style="text-decoration: none;"
          >
            <button
              class="the-button"
            >
              <div
                class="the-circle"
              />
              <div
                class="vertical-bar"
              />
              <div
                class="horizontal-bar"
              />
            </button>
          </a>
          <p
            class="label"
          >
            Add a Book
          </p>
        </div>
        <div
          class="small-button"
        >
          <div />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="footer-grand-wrapper"
    >
      <div
        class="small-button"
      >
        <div />
      </div>
      <div
        class="circle-add-button-wrapper"
      >
        <a
          href="/findadd"
          style="text-decoration: none;"
        >
          <button
            class="the-button"
          >
            <div
              class="the-circle"
            />
            <div
              class="vertical-bar"
            />
            <div
              class="horizontal-bar"
            />
          </button>
        </a>
        <p
          class="label"
        >
          Add a Book
        </p>
      </div>
      <div
        class="small-button"
      >
        <div />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Welcome Text should render based on props passed should render with two corner buttons 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="footer-grand-wrapper"
      >
        <div
          class="small-button"
        >
          <a
            href="/profile/"
            style="text-decoration: none;"
          >
            <div
              class="user-button-wrapper"
            >
              <button
                class="user-button"
              />
              <p>
                User Home
              </p>
            </div>
          </a>
        </div>
        <div />
        <div
          class="small-button"
        >
          <a
            href="/search"
            style="text-decoration: none;"
          >
            <div
              class="search-button-wrapper"
            >
              <button
                class="search-button"
              />
              <p>
                Find a Book
              </p>
            </div>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="footer-grand-wrapper"
    >
      <div
        class="small-button"
      >
        <a
          href="/profile/"
          style="text-decoration: none;"
        >
          <div
            class="user-button-wrapper"
          >
            <button
              class="user-button"
            />
            <p>
              User Home
            </p>
          </div>
        </a>
      </div>
      <div />
      <div
        class="small-button"
      >
        <a
          href="/search"
          style="text-decoration: none;"
        >
          <div
            class="search-button-wrapper"
          >
            <button
              class="search-button"
            />
            <p>
              Find a Book
            </p>
          </div>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
